src/masternodes/validation.cpp

```diff
@@ -2428,6 +2428,11 @@ static void ProcessEVMQueue(const CBlock &block, const CBlockIndex *pindex, CCus
         }
 
         const auto blockResult = evm_finalize(evmContext, false, block.nBits, beneficiary, block.GetBlockTime());
+        auto evmBlockHash = std::vector<uint8_t>(blockResult.block_hash.begin(), blockResult.block_hash.end());
+        std::reverse(evmBlockHash.begin(), evmBlockHash.end());
+
+        pcustomcsview->SetBlockHash(CEvmDvmMapType::DvmEvm, block.GetHash(), uint256(evmBlockHash));
```

Comment:
Please use cache instead of pcustomcsview, this is a copy of CCustomCSView which will be added to the block undo.